wal/wal.go

```diff
@@ -261,6 +261,8 @@ func (w *WAL) Repair(err error) error {
 	// But that's not generally applicable if the records have any kind of causality.
 	// Maybe as an extra mode in the future if mid-WAL corruptions become
 	// a frequent concern.
+	err = errors.Cause(err) // So that we can pick up errors even if wrapped.
+
 	cerr, ok := err.(*CorruptionErr)
 	if !ok {
 		return errors.New("cannot handle error")
```

Comment:
This should probably show the original error as well, i.e. `errors.Wrap(origErr, "cannot handle error")`